src/include/ipc4/gateway.h

```diff
@@ -204,4 +204,6 @@ enum ipc4_direction_type {
 };
 
 #define IPC4_DIRECTION(x) BIT(x)
+
+#define IPC4_MAX_MULTIPLE_GTW_COUNT 8
```

Comment:
come on guys, can you please talk?

This is defined as 16 by @bardliao in https://github.com/thesofproject/linux/pull/3580/commits/2c038564e95f85796d85f74deffc79084f6bf1a9#diff-d8bbc05d879b6eee2041d6fc0ee06f050be097ac05b12cfec9b35d89f66d3a84R43.

In addition there's a magic value of 0x50 to make the different between 'regular' and 'multi' gateways.

Reply:
max gtw count is different with multiple gtw count which is for single stream with multiple gateway support. we can't use 16 alh codec tos support one stream.
0x50 is not used by fw since we get data directly from mapping table

Reply:
Discussed with @bardliao  and will use dynamic allocation for multiple gateway count.
For node id < 0x50 case, there may be none of mapping table, so we have to reuse node id.
I will update this PR
````
{
      node_id  \\ multiple gtw id if >= 0x50, normal gtw id < 0x50
     {
          count;
         {
            node_id; \\ normal gtw id,  this is the finally id
            mask
         }  mapping[MAX_COUNT]
      }
}
````

Reply:
The logic is:
(1) if count > 0,  get node id from mapping table
(2) else get node id from gtw_node_id. In this case, gtw_node_id should < magic number of 50. 
So FW doesn't need 0x50 except validation like "assert(gtw_node_id >= 0x50)" for multiple gtw case

For kernel driver, it will used 0x50 since it has to used it for close source fw.